### PR TITLE
Add README and CONTRIBUTING

### PR DESCRIPTION
**Pull request by simonswine**
This is my first stab at both a README and a CONTRIBUTING markdown. I have used the cortex contributing guide as source of inspiration :slightly_smiling_face:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Grafana Dskit
+
+This library contains utilities that are useful for building distributed
+services.
+
+## Current state
+
+This library is still in development. In a first stage we plan to move over
+utilities from the [Cortex] project.
```
Comment (aknuds1) [inline README.md:9]: Nit, but I think "during the first stage" sounds more precise:
```suggestion
This library is still in development. During the first stage we plan to move over
utilities from the [Cortex] project.
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+## Contributing
+
+If you're interested in contributing to this project:
+
+- Start by reading the [Contributing guide](/CONTRIBUTING.md).
+
+## License
+
+[Apache 2.0 License](https://github.com/grafana/dskit/blob/main/LICENSE)
+
```
Comment (aknuds1) [inline README.md:22]: Nit, but can we remove this extra newline?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+To run the unit tests suite:
+
+```bash
+make test
+```
+
+### Dependency management
+
+We use [go modules] to manage dependencies on external packages.  This requires
```
Comment (aknuds1) [inline CONTRIBUTING.md:47]: Nit, just spotting an extra space.
```suggestion
We use [go modules] to manage dependencies on external packages. This requires
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+a working Go environment with version 1.16 or greater, git and [bzr] installed.
+
+[Go modules]: https://golang.org/cmd/go/#hdr-Modules__module_versions__and_more
+[bzr]: http://wiki.bazaar.canonical.com/Download
```
Comment (pstibrany) [inline CONTRIBUTING.md:51]: :) this looks old

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+go get example.com/some/module/pkg
+
+# Pick a specific version.
+go get example.com/some/module/pkg@vX.Y.Z
+```
+
+Tidy up the `go.mod` and `go.sum` files:
+
+```bash
+go mod tidy
```
Comment (pstibrany) [inline CONTRIBUTING.md:66]: should we include `go mod vendor` here too? Otherwise lint check will fail.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+You have to commit the changes to `go.mod` and `go.sum` before submitting the
+pull request.
+
+Dskit uses the `goimports` tool (`go get golang.org/x/tools/cmd/goimports` to
+install) to format the Go files, and sort imports. We use goimports with
+`-local github.com/grafana/dskit` parameter, to put Dskit internal imports into
+a separate group. We try to keep imports sorted into three groups:
+imports from standard library, imports of 3rd party packages and internal
+Cortex imports. Goimports will fix the order, but will keep existing newlines
```
Comment (aknuds1) [inline CONTRIBUTING.md:79]: I think there's a stray Cortex reference there.
```suggestion
a separate group. We try to keep imports sorted into three groups:
imports from standard library, imports of 3rd party packages and internal
imports. Goimports will fix the order, but will keep existing newlines
```

**Review by pstibrany — APPROVED** [review]

**Review by aknuds1 — CHANGES_REQUESTED** [review]
Thank you for adding these, they generally look good. Apart from nits, I just think you should fix what looks like a stray reference to Cortex in the contribution guide.

File: CONTRIBUTING.md (hunk 4 above)
Comment (simonswine) [reply CONTRIBUTING.md:51]: Agreed and removed, haven't heard about bazaar in a while :joy:

File: CONTRIBUTING.md (hunk 5 above)
Comment (simonswine) [reply CONTRIBUTING.md:65]: We currently don't have a vendor folder checked in and hence we also don't do a lint check on it.

https://github.com/grafana/dskit/blob/4710207529678a6e5f8954d1c280b1d099a3a70e/Makefile#L27-L32

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+To run the unit tests suite:
+
+```bash
+make test
+```
+
+### Dependency management
+
+We use [go modules] to manage dependencies on external packages. This requires
```
Comment (aknuds1) [inline CONTRIBUTING.md:47]: Total nit, but looks as if we mostly write it as _Go_:
```suggestion
We use [Go modules] to manage dependencies on external packages. This requires
```

File: CONTRIBUTING.md (hunk 5 above)
Comment (aknuds1) [reply CONTRIBUTING.md:65]: Christian is right I think.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+    ```
+    # Ubuntu/Debian
+    apt-get install git-filter-repo
+
+    # MacOSX
+    brew install git-filter-repo
+    ```
+
+### Import a particular files
```
Comment (aknuds1) [inline CONTRIBUTING.md:95]:
```suggestion
### Import particular files
```

**Review by aknuds1 — APPROVED** [review]
LGTM!

File: CONTRIBUTING.md (hunk 5 above)
Comment (pstibrany) [reply CONTRIBUTING.md:65]: I see. I thought we follow the same practice as in Cortex, but forgot to check. Thanks!

**Review by pstibrany — APPROVED** [review]

**Pull request merged**